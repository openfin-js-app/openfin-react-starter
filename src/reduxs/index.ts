import {combineReducers} from 'redux';
import {IApplicationState} from './application/types';
import {IClientState} from './client/types';
import {IConfigState} from './config/types';

import application from './application/reducer';
import { defaultState as applicationDefaultState } from './application/reducer';
import client from './client/reducer';
import { defaultState as clientDefaultState } from './client/reducer';
import config from './config/reducer';
import { defaultState as configDefaultState } from './config/reducer';

export * from './application/actions';
export * from './application/types';
export * from './client/actions';
export * from './client/types';
export * from './config/actions';
export * from './config/context';
export * from './config/types';

export interface IRootState {
    application:IApplicationState,
    client:IClientState,
    config:IConfigState,
}

export const rootDefaultState = {
    application:applicationDefaultState,
    client:clientDefaultState,
    config:configDefaultState,
}

export default (parentWindowState?:IRootState)=>{
    if (parentWindowState){
        return combineReducers({
            application:application(parentWindowState.application),
            client:client(parentWindowState.client),
            config:config(parentWindowState.config),
        });

    }else{
        return combineReducers({
            application:application(),
            client:client(),
            config:config(),
        });
    }
}
