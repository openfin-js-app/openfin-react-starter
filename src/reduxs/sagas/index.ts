import { all, takeEvery, select } from 'redux-saga/effects';

import clientSaga from './client';

export function* handleLogAllActions(action) {
    const state = yield select();
    console.log(`${action.type}`,action,'state after',state);
}

export function* watchAndLogSaga(){
    yield takeEvery('*',handleLogAllActions)
}

export default function* rootSaga(){

    const sagas = [ clientSaga() ];

    if(process.env.REACT_APP_LOG_ACTION === 'true'){
        sagas.unshift(watchAndLogSaga());
    }

    yield all(sagas);

}