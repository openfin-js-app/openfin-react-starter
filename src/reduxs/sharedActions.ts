import {
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
} from './application/actions';

import {
    CLIENT_SET_VALUE,
} from './client/actions';

const sharedActions:string[]=[
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
    CLIENT_SET_VALUE,
];

export default sharedActions;