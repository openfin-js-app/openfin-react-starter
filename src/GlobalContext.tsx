import * as React from 'react';
import { connect } from 'react-redux'

import { I18Language, IConfigRuntimeState } from './reduxs'

import { ConfigContextProvider } from './reduxs/config/context';

interface IProps {
    config:IConfigRuntimeState,
    onToggleThemeField?:()=>void,
    onUpdateLangField?:(lang:I18Language)=>void,
}

class GlobalContext extends React.Component<IProps,{}>{

    render(){

        const {
            config,
            onToggleThemeField,
            onUpdateLangField,
        } = this.props;

        return (
            <ConfigContextProvider value={
                {
                    config,
                    actions:{
                        onToggleThemeField,
                        onUpdateLangField,
                    }
                }
            }>
                {this.props.children}
            </ConfigContextProvider>

        )
    }
}

export default GlobalContext;