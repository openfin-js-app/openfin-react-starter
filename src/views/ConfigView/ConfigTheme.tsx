import * as React from 'react';
import { WithStyles, withStyles } from '@material-ui/core/styles';

import Switch from '@material-ui/core/Switch';
import Typography from '@material-ui/core/Typography';

import cx from "classnames";
import { connect } from 'react-redux';

import { withNamespaces, WithNamespaces } from 'react-i18next';

import { configThemeViewStyle as style } from '../../assets/jss/openfin-starter';

import {
    MuiTheme,
} from '../../reduxs'

import {
    WithConfigContext,
    withConfigContext
} from '../../reduxs/config/context'

interface IProps extends WithStyles<typeof style>, WithNamespaces,WithConfigContext{
    actions:{
    },
}

import { IRootState } from '../../reduxs';

class ConfigThemeView extends React.Component<IProps,{}>{
    render(){

        const {
            classes, t,
            configContext:{
                config,
                actions:{
                    handleToggleThemeField
                }
            }
        } = this.props;

        return (
            <div className={classes.container}>
                <Switch
                    checked={config.application.theme === MuiTheme.DARK}
                    onChange={handleToggleThemeField}
                    value="themeVal"
                />
                <Typography className={classes.themeSpan} variant="body1" gutterBottom>
                    {t(`common.${config.application.theme}`)}
                </Typography>
            </div>
        )
    }
}

export default connect(
    (state:IRootState)=>({
    }),
    dispatch => ({
        actions:{
        }
    }),
    )(
    withStyles(style)(
        withNamespaces('config')(withConfigContext(ConfigThemeView))
    )
);

