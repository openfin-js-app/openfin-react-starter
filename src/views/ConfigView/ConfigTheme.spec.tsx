import * as React from 'react';
import { Provider } from 'react-redux';

import Switch from '@material-ui/core/Switch';

import { createShallow, createMount } from '@material-ui/core/test-utils';
import configurestore from 'redux-mock-store';

import ConfigTheme from './ConfigTheme';

import GlobalContext from '../../GlobalContext';
import {MuiTheme} from "../../reduxs";

const mockStore = configurestore();

const darkState = {
    config:{
        application:{
            theme:MuiTheme.DARK
        },
    },
};

const lightState = {
    config:{
        application:{
            theme:MuiTheme.LIGHT
        },
    },
};

describe('Config Theme',()=>{

    let shallow;
    let mount;

    beforeEach(() => {
        mount = createMount();
        shallow = createShallow();
    });

    afterEach(()=>{
        mount.cleanUp();
    });

    it('render dark correctly and switch to light',()=>{

        const store = mockStore(darkState);
        const handleToggleThemeField = jest.fn();
        const wrapper = mount(
            <Provider store={store}>
                <GlobalContext
                    config={darkState.config}
                    onToggleThemeField={handleToggleThemeField}
                >
                    <ConfigTheme/>
                </GlobalContext>
            </Provider>
        );

        const switches = wrapper.find(Switch);
        expect(switches).toHaveLength(1);

        switches.at(0).props().onChange();

        expect(handleToggleThemeField).toHaveBeenCalled();
        expect(handleToggleThemeField.mock.calls).toMatchSnapshot();


    })

    it('render light correctly and switch to dark',()=>{

        const store = mockStore(lightState);
        const handleToggleThemeField = jest.fn();
        const wrapper = mount(
            <Provider store={store}>
                <GlobalContext
                    config={lightState.config}
                    onToggleThemeField={handleToggleThemeField}
                >
                    <ConfigTheme/>
                </GlobalContext>
            </Provider>
        );

        const switches = wrapper.find(Switch);
        expect(switches).toHaveLength(1);

        switches.at(0).props().onChange();

        expect(handleToggleThemeField).toHaveBeenCalled();
        expect(handleToggleThemeField.mock.calls).toMatchSnapshot();


    })

})