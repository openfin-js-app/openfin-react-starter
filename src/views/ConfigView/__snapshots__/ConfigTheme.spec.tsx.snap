// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigLang render dark correctly and switch to light 1`] = `
Array [
  Array [],
]
`;

exports[`ConfigLang render light correctly and switch to dark 1`] = `
Array [
  Array [],
]
`;
